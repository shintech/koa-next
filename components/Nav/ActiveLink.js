import { withRouter } from 'next/router'
import React, { Children } from 'react'
import routes, { Link } from 'routes'

const ActiveLink = ({ router, children, ...props }) => {
  const child = Children.only(children)
  const href = routes.findAndGetUrls(props.route, props.params).urls.as

  let className = child.props.className || null

  if (router.pathname === href && props.activeClassName) {
    className = `${className !== null ? className : ''} ${props.activeClassName}`.trim()
  }

  const onClick = (className === 'active') ? e => { e.preventDefault() } : null

  delete props.activeClassName

  return <Link {...props}>{React.cloneElement(child, { className, onClick })}</Link>
}

export default withRouter(ActiveLink)
