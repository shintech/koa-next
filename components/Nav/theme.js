export default options => ({
  background: 'DarkMagenta',
  text: 'gainsboro',
  tagBackground: 'DarkMagenta',
  svg: 'gold',
  active: 'teal',
  activeText: 'darkslategrey',
  activeHover: 'steelblue',
  activeHoverText: 'black',
  hover: 'violet',
  hoverText: 'black',
  activeSVG: 'black'
})
