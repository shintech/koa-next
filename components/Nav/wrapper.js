import styled from 'styled-components'

const Wrapper = styled.nav`
  grid-row: span 3;
  grid-column: 1;

  display: flex;
  flex-direction: column;

  overflow: hidden;

  background-color: ${props => props.theme.background};

  .icon {
    display: none;
  }

  a {
    color: ${props => props.theme.text};
    background-color: ${props => props.theme.tagBackground};
    text-align: center;
    display: flex;
    align-items: center;

    padding: 1ch;
    text-decoration: none;
    font-size: 2.5ch;

    border-color: black;
    border-style: solid;
    border-width: 0 0 1px 0;
    border-radius: 1ch 0 0 1ch;

    &.active {
      background-color: ${props => props.theme.active};
      color: ${props => props.theme.activeText};

      &:hover{
        background-color: ${props => props.theme.activeHover};
        color: ${props => props.theme.activeHoverText};
        cursor: not-allowed;
      }
    }

    :hover {
      background-color: ${props => props.theme.hover};
      color: ${props => props.theme.hoverText};
    }
  }

  svg.logo {
    height: 9ch;
    width: 9ch;
    margin: auto 2ch 0 auto;
  }

  @media (max-width: 800px) {
    grid-row: 1;
    grid-column: span 3;
    flex-direction: row;
    border-radius: 1ch;
    align-items: center;

    a {
      border-radius: 0;
      border: none;

      :first-child {
        border-radius: 1ch 0 0 1ch;
      }

      &:not(:first-child) {display: none;}
      &.icon {
        margin-left: auto;
        display: block;
      }
    }

    svg.logo {
      display: none;
    }
    
    &.responsive {
      flex-direction: column;
      position: relative;

      .icon {
        position: absolute;
        right: 0;
        top: 0;
      }      
      
      a {
        display: block;
        
        margin-left: 3.5ch;
        
        :last-child {
          border-radius: 0 1ch 1ch 0;
        }
        
        :not(:last-child) {
          width: 100%;
          text-align: left;
        }
      }
    }
  }
`

export default Wrapper
