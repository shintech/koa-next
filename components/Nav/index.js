import React from 'react'
import Link from './ActiveLink'
// import PropTypes from 'prop-types'
import Wrapper from './wrapper'
import theme from './theme'
import SVG from 'static/images/nodejs.svg'
import Router from 'routes'

function toggleResponsive () { // TODO create redux action
  var x = document.getElementById('container')
  if (x.classList.contains('responsive')) {
    x.classList.remove('responsive')
  } else {
    x.classList.add('responsive')
  }
}

const Nav = () =>
  <Wrapper id='container'theme={theme()}>
    {Router.routes.map(link => {
      let name = (link.name === 'index') ? 'home' : link.name

      return <Link key={name} prefetch activeClassName='active' route={link.page}><a>{name}</a></Link>
    })}

    <SVG className='logo' />

    <a href='javascript:void(0);' className='icon' onClick={() => { toggleResponsive() }}>
      <i className='fa fa-bars' />
    </a>
  </Wrapper>

Nav.propTypes = {}

export default Nav
