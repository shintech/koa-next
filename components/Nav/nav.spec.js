/* eslint-env jest */

import { shallow } from 'enzyme'
import { compose } from 'redux'
import toJSON from 'enzyme-to-json'
import Nav from 'components/Nav'

const shallowExpect = compose(expect, toJSON, shallow)

describe('COMPONENT -> Nav - snapshot...', () => {
  it('expect to render correct properties', () => {
    shallowExpect(<Nav />).toMatchSnapshot()
  })
})
