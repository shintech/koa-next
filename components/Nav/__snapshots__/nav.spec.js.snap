// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT -> Nav - snapshot... expect to render correct properties 1`] = `
<wrapper__Wrapper
  id="container"
  theme={
    Object {
      "active": "teal",
      "activeHover": "steelblue",
      "activeHoverText": "black",
      "activeSVG": "black",
      "activeText": "darkslategrey",
      "background": "DarkMagenta",
      "hover": "violet",
      "hoverText": "black",
      "svg": "gold",
      "tagBackground": "DarkMagenta",
      "text": "gainsboro",
    }
  }
>
  <withRouter(ActiveLink)
    activeClassName="active"
    key="home"
    prefetch={true}
    route="/"
  >
    <a>
      home
    </a>
  </withRouter(ActiveLink)>
  <withRouter(ActiveLink)
    activeClassName="active"
    key="about"
    prefetch={true}
    route="/about"
  >
    <a>
      about
    </a>
  </withRouter(ActiveLink)>
  <withRouter(ActiveLink)
    activeClassName="active"
    key="contact"
    prefetch={true}
    route="/contact"
  >
    <a>
      contact
    </a>
  </withRouter(ActiveLink)>
  <Component
    className="logo"
  />
  <a
    className="icon"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <i
      className="fa fa-bars"
    />
  </a>
</wrapper__Wrapper>
`;
